osquery/tables/system/darwin/processes.cpp

```diff
@@ -314,6 +315,29 @@ void genProcArch(QueryContext& context, int pid, ProcessesRow& r) {
     r.cpu_type_col = -1;
     r.cpu_subtype_col = -1;
   }
+
+  if (archinfo.p_cputype == CPU_TYPE_X86_64) {
+    r.kind_col = "intel";
+  }
+
+  if (archinfo.p_cputype == CPU_TYPE_ARM64) {
+    r.kind_col = "apple";
+
+    struct kinfo_proc kinfo {};
+    int mib[] = {CTL_KERN, KERN_PROC, KERN_PROC_PID, pid};
+    size_t size = sizeof(kinfo);
+
+    if (sysctl(mib, 4, &kinfo, &size, nullptr, 0) != 0 ||
+        size < sizeof(kinfo)) {
+      return;
+    }
+
+    // proc_bsdinfo also has pbi_flags, but that seems to be not always
+    // populated, instead kinfo_proc works better to get at the process flags
+    if (kinfo.kp_proc.p_flag & P_TRANSLATED) {
+      r.kind_col = "intel";
+    }
```

Comment:
So this overrides the earlier `r.kind_col` settings? That feels like we're trying to encode logic about how the OS handles things.

What about exposing `translated` as a boolean column?

Reply:
Went with `translated` as a boolean column!